src/rust/src/x509/sign.rs

```diff
@@ -256,3 +259,265 @@ pub(crate) fn sign_data<'p>(
     };
     signature.extract()
 }
+
+fn py_hash_from_hash_type(hash_type: HashType) -> Option<&'static str> {
+    match hash_type {
+        HashType::None => None,
+        HashType::Sha224 => Some("SHA224"),
+        HashType::Sha256 => Some("SHA256"),
+        HashType::Sha384 => Some("SHA384"),
+        HashType::Sha512 => Some("SHA512"),
+        HashType::Sha3_224 => Some("SHA3_224"),
+        HashType::Sha3_256 => Some("SHA3_256"),
+        HashType::Sha3_384 => Some("SHA3_384"),
+        HashType::Sha3_512 => Some("SHA3_512"),
+    }
+}
+
+pub(crate) fn verify_signature_with_oid<'p>(
+    py: pyo3::Python<'p>,
+    issuer_public_key: &'p pyo3::PyAny,
+    signature_oid: &asn1::ObjectIdentifier,
+    signature: &[u8],
+    data: &[u8],
+) -> PyAsn1Result<&'p pyo3::PyAny> {
```

Comment:
```suggestion
) -> PyAsn1Result<()> {
```

No need for this to return anything